metadata-ingestion/src/datahub/ingestion/source/pulsar.py

```diff
@@ -89,7 +89,16 @@ def __init__(self, schema):
             logger.error(f"Invalid JSON schema: {schema_data}. Error: {str(e)}")
             avro_schema = {}
 
-        self.schema_name = avro_schema.get("namespace") + "." + avro_schema.get("name")
+        namespace = avro_schema.get("namespace")
+        name = avro_schema.get("name")
+        if not namespace:
+            logger.warning("namespace is missing in schema, using 'default_namespace'")
+            namespace = "default_namespace"
+        if not name:
+            logger.warning("name is missing in schema, using 'default_name'")
+            name = "default_name"
+        self.schema_name = namespace + "." + name
+
```

Comment:
Checking avro specs, both `namespace` and `name` are optional. 
Only `name` is required in the case of `record` or `enum` and optional if _eg_ primitive types or arrays.
So the fix makes sense to cover that scenario.

I'm concerned about having such a fallback values, since they do not match the original avro schema definition.

Ideally, if both `namespace` and `name` are missed, `schema_name` should be null or empty string.
This `schema_name` is used as the `schemaName` for the `SchemaMetadataKey`, which is required and len > 1.

https://github.com/datahub-project/datahub/blob/b091e4615d915783818c1750599723b933d41267/metadata-models/src/main/pegasus/com/linkedin/schema/SchemaMetadataKey.pdl#L10-L17

So what about something like that:

```suggestion
        self.schema_name = "null"
        if avro_schema.get("namespace") and avro_schema.get("name"):
            self.schema_name = avro_schema.get("namespace") + "." + avro_schema.get("name")
        elif avro_schema.get("namespace"):
            self.schema_name = avro_schema.get("namespace")
        elif avro_schema.get("name"):
            self.schema_name = avro_schema.get("name")
```

WDYT @hsheth2 ? Other solutions may require to update the `SchemaMetadataKey` definition.

Reply:
yeah that seems pretty reasonable

we don't really use the schema_name for much, so a fallback value of `null` feels reasonable over throwing an exception / reporting a warning